go/client/logclient.go

```diff
@@ -179,20 +180,36 @@ func (c *LogClient) postAndParse(uri string, req interface{}, res interface{}) (
 }
 
 // Attempts to add |chain| to the log, using the api end-point specified by
-// |path|.
-func (c *LogClient) addChainWithRetry(path string, chain []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error) {
+// |path|. If provided deadline is passed before submission is complete an
+// error will be returned.
+func (c *LogClient) addChainWithRetry(path string, chain []ct.ASN1Cert, deadline context.Context) (*ct.SignedCertificateTimestamp, error) {
 	var resp addChainResponse
 	var req addChainRequest
 	for _, link := range chain {
 		req.Chain = append(req.Chain, base64.StdEncoding.EncodeToString(link))
 	}
-	done := false
 	httpStatus := "Unknown"
+	backoffSeconds := 0
+	done := false
 	for !done {
-		backoffSeconds := 0
+		backoffTimer := time.NewTimer(0)
```

Comment:
Timer provides a channel which can be checked concurrently with the
context, unlike a Sleep which just blocks everything

On Tue, Jan 26, 2016 at 9:29 PM Jeff Hodges <notifications@github.com>
wrote:

> In go/client/logclient.go
> <https://github.com/letsencrypt/certificate-transparency/pull/1#discussion_r50942953>
> :
>
> >  	for !done {
> > -		backoffSeconds := 0
> > +		backoffTimer := time.NewTimer(0)
>
> So, why the switch to NewTimer from a time.Sleep?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/letsencrypt/certificate-transparency/pull/1/files#r50942953>
> .
>